src/lib/contracts.ts

```diff
@@ -198,67 +241,45 @@ export function validateContract(contract: unknown): boolean {
 
 	return true;
 }
+
 export function validateTargetContracts(
 	apps: TargetApps,
-): Dictionary<ApplicationContractResult> {
-	return Object.keys(apps)
-		.map((appUuid): [string, ApplicationContractResult] => {
-			const app = apps[appUuid];
-			const [release] = Object.values(app.releases);
-			const serviceContracts = Object.keys(release?.services ?? [])
-				.map((serviceName) => {
-					const service = release.services[serviceName];
-					const { contract } = service;
-					if (contract) {
-						try {
-							// Validate the contract syntax
-							validateContract(contract);
-
-							return {
-								serviceName,
-								contract,
-								optional: checkTruthy(
-									service.labels?.['io.balena.features.optional'],
-								),
-							};
-						} catch (e: any) {
-							throw new ContractValidationError(serviceName, e.message);
+): Dictionary<AppContractResult> {
+	const result: Dictionary<AppContractResult> = {};
+
+	for (const [appUuid, app] of Object.entries(apps)) {
+		// While app.releases is an object, we expect a target to only
+		// contain a single release per app so we use the first result of
+		// the map call
+		const [appContractResult] = Object.entries(app.releases).map(
+			([commit, release]) => {
```

Comment:
The code seems to imply there's exactly 0 or 1 entries, so this could be more like
```suggestion
		if (app.releases.length === 0) {
			return;
		}
		const [commit, release] = app.releases[0]
```
(and could optionally have a check if it's >1 and fail in that case. As it stands it looks like it implies it can handle arbitrary lengths due to the `.map` when in practice it really doesn't expect that

Reply:
Yep, the code is cleaner in this way. Done